File: library/src/main/java/com/hjhrq1991/library/WebChromeClientListener.java
package com.hjhrq1991.library;

import android.graphics.Bitmap;
import android.net.Uri;
import android.view.View;
import android.webkit.ConsoleMessage;
import android.webkit.GeolocationPermissions;
import android.webkit.JsPromptResult;
import android.webkit.JsResult;
import android.webkit.PermissionRequest;
import android.webkit.ValueCallback;
import android.webkit.WebChromeClient;
import android.webkit.WebStorage;
import android.webkit.WebView;

/**
 * @author hjhrq1991 created at 16/11/21 10 38.
 */
public class WebChromeClientListener implements OnWebChromeClientListener {

    @Override
    public void onReceivedTitle(WebView view, String title) {

    }

    @Override
    public void onProgressChanged(WebView view, int newProgress) {

    }

    @Override
    public void onReceivedIcon(WebView view, Bitmap icon) {

    }

    @Override
    public void onReceivedTouchIconUrl(WebView view, String url, boolean precomposed) {

    }

    @Override
    public void onShowCustomView(View view, WebChromeClient.CustomViewCallback callback) {

    }

    @Override
    public void onShowCustomView(View view, int requestedOrientation, WebChromeClient.CustomViewCallback callback) {

    }

    @Override
    public void onHideCustomView() {

    }

    @Override
    public boolean onCreateWindow(WebView view, boolean isDialog, boolean isUserGesture, android.os.Message resultMsg) {
        return false;
    }

    @Override
    public void onRequestFocus(WebView view) {

    }

    @Override
    public void onCloseWindow(WebView window) {

    }

    @Override
    public boolean onJsAlert(WebView view, String url, String message, JsResult result) {
        return false;
    }

    @Override
    public boolean onJsConfirm(WebView view, String url, String message, JsResult result) {
        return false;
    }

    @Override
    public boolean onJsPrompt(WebView view, String url, String message, String defaultValue, JsPromptResult result) {
        return false;
    }

    @Override
    public boolean onJsBeforeUnload(WebView view, String url, String message, JsResult result) {
        return false;
    }

    @Override
    public void onExceededDatabaseQuota(String url, String databaseIdentifier, long quota, long estimatedDatabaseSize, long totalQuota, WebStorage.QuotaUpdater quotaUpdater) {

    }

    @Override
    public void onReachedMaxAppCacheSize(long requiredStorage, long quota, WebStorage.QuotaUpdater quotaUpdater) {

    }

    @Override
    public void onGeolocationPermissionsShowPrompt(String origin, GeolocationPermissions.Callback callback) {

    }

    @Override
    public void onGeolocationPermissionsHidePrompt() {

    }

    @Override
    public void onPermissionRequest(PermissionRequest request) {

    }

    @Override
    public void onPermissionRequestCanceled(PermissionRequest request) {

    }

    @Override
    public boolean onJsTimeout() {
        return false;
    }

    @Override
    public void onConsoleMessage(String message, int lineNumber, String sourceID) {

    }

    @Override
    public boolean onConsoleMessage(ConsoleMessage consoleMessage) {
        return false;
    }

    @Override
    public Bitmap getDefaultVideoPoster() {
        return null;
    }

    @Override
    public View getVideoLoadingProgressView() {
        return null;
    }

    @Override
    public void getVisitedHistory(ValueCallback<String[]> callback) {

    }

    @Override
    public boolean onShowFileChooser(WebView webView, ValueCallback<Uri[]> filePathCallback, WebChromeClient.FileChooserParams fileChooserParams) {
        return false;
    }
}

File: library_ktx/src/main/java/com/hjhrq1991/library/WebChromeClientListener.kt
package com.hjhrq1991.library

import android.graphics.Bitmap
import android.net.Uri
import android.view.View
import android.webkit.ConsoleMessage
import android.webkit.GeolocationPermissions
import android.webkit.JsPromptResult
import android.webkit.JsResult
import android.webkit.PermissionRequest
import android.webkit.ValueCallback
import android.webkit.WebChromeClient
import android.webkit.WebStorage
import android.webkit.WebView

/**
 * @author hjhrq1991 created at 16/11/21 10 38.
 */
open class WebChromeClientListener : OnWebChromeClientListener {
    override fun onReceivedTitle(view: WebView?, title: String?) {
    }

    override fun onProgressChanged(view: WebView?, newProgress: Int) {
    }

    override fun onReceivedIcon(view: WebView?, icon: Bitmap?) {
    }

    override fun onReceivedTouchIconUrl(view: WebView?, url: String?, precomposed: Boolean) {
    }

    override fun onShowCustomView(view: View?, callback: WebChromeClient.CustomViewCallback?) {
    }

    override fun onShowCustomView(view: View?, requestedOrientation: Int, callback: WebChromeClient.CustomViewCallback?) {
    }

    override fun onHideCustomView() {
    }

    override fun onCreateWindow(view: WebView?, isDialog: Boolean, isUserGesture: Boolean, resultMsg: android.os.Message?): Boolean {
        return false
    }

    override fun onRequestFocus(view: WebView?) {
    }

    override fun onCloseWindow(window: WebView?) {
    }

    override fun onJsAlert(view: WebView?, url: String?, message: String?, result: JsResult?): Boolean {
        return false
    }

    override fun onJsConfirm(view: WebView?, url: String?, message: String?, result: JsResult?): Boolean {
        return false
    }

    override fun onJsPrompt(view: WebView?, url: String?, message: String?, defaultValue: String?, result: JsPromptResult?): Boolean {
        return false
    }

    override fun onJsBeforeUnload(view: WebView?, url: String?, message: String?, result: JsResult?): Boolean {
        return false
    }

    override fun onExceededDatabaseQuota(url: String?, databaseIdentifier: String?, quota: Long, estimatedDatabaseSize: Long, totalQuota: Long, quotaUpdater: WebStorage.QuotaUpdater?) {
    }

    override fun onReachedMaxAppCacheSize(requiredStorage: Long, quota: Long, quotaUpdater: WebStorage.QuotaUpdater?) {
    }

    override fun onGeolocationPermissionsShowPrompt(origin: String?, callback: GeolocationPermissions.Callback?) {
    }

    override fun onGeolocationPermissionsHidePrompt() {
    }

    override fun onPermissionRequest(request: PermissionRequest?) {
    }

    override fun onPermissionRequestCanceled(request: PermissionRequest?) {
    }

    override fun onJsTimeout(): Boolean {
        return false
    }

    override fun onConsoleMessage(message: String?, lineNumber: Int, sourceID: String?) {
    }

    override fun onConsoleMessage(consoleMessage: ConsoleMessage?): Boolean {
        return false
    }

    override fun getDefaultVideoPoster(): Bitmap? {
        return null
    }

    override fun getVideoLoadingProgressView(): View? {
        return null
    }

    override fun getVisitedHistory(callback: ValueCallback<Array<String>>?) {
    }

    override fun onShowFileChooser(webView: WebView?, filePathCallback: ValueCallback<Array<Uri>>?, fileChooserParams: WebChromeClient.FileChooserParams?): Boolean {
        return false
    }
}
